File: videostore/csharp/src/MovieRental/Statements/Customer.cs
using System.Collections.Generic;

namespace MovieRental.Statements
{
    public class Customer
    {
        private List<Rental> _rentals = new List<Rental>();

        public Customer(string name)
        {
            this.name = name;
        }

        public string name { get; set; }
        
        public List<Rental> rentals
        {
            get { return _rentals; }
            set { _rentals = value; }
        }

        public void AddRental(Rental rental)
        {
            rentals.Add(rental);
        }

        public string Statement()
        {
            double totalAmount = 0.0;
            int frequentRenterPoints = 0;
            string result = "Rental Record for " + name + "\n";

            foreach (var rental in rentals)
            {
                double thisAmount = 0;

                switch (rental.Movie.PriceCode)
                {
                    case Movie.REGULAR:
                        thisAmount += 2;
                        if (rental.DaysRented > 2)
                            thisAmount += (rental.DaysRented - 2)*1.5;
                        break;
                    case Movie.NEW_RELEASE:
                        thisAmount += rental.DaysRented*3;
                        break;
                    case Movie.CHILDRENS:
                        thisAmount += 1.5;
                        if (rental.DaysRented > 3)
                            thisAmount += (rental.DaysRented - 3)*1.5;
                        break;
                }

                frequentRenterPoints++;

                if (rental.Movie.PriceCode == Movie.NEW_RELEASE && rental.DaysRented > 1)
                    frequentRenterPoints++;

                result += "\t" + rental.Movie.Title + "\t" + thisAmount.ToString() + "\n";
                totalAmount += thisAmount;
            }

            result += "You owed " + totalAmount.ToString() + "\n";
            result += "You earned " + frequentRenterPoints.ToString() + " frequent renter points\n";

            return result;
        }
    }
}

File: videostore/java/src/main/java/domain/Customer.java
package domain;

import java.util.ArrayList;
import java.util.List;

public class Customer {

    private String name;
    private List<Rental> rentals = new ArrayList<Rental>();

    public Customer(String name) {
        this.name = name;
    }

    public void addRental(Rental rental) {
        rentals.add(rental);
    }

    public String getName() {
        return name;
    }

    double totalAmount = 0;
    int frequentRenterPoints = 0;

    public String statement() {
        return getHeader() +
                getLines() +
                getFooter();
    }

    private String getHeader() {
        return "Rental Record for " + getName() + "\n";
    }

    private String getFooter() {
        return "You owed " + totalAmount +
                "\nYou earned " + frequentRenterPoints + " frequent renter points\n";
    }

    private String getLines() {
        String lines = "";
        for (Rental each : rentals) {
            lines += getLine(each);
        }
        return lines;
    }

    private String getLine(Rental each) {
        double thisAmount = each.calculatePrice();
        frequentRenterPoints += each.calculatePoints();
        totalAmount += thisAmount;
        return "\t" + each.getMovie().getTitle() + "\t"
                + thisAmount + "\n";
    }


}